mitodl/ocw-studio
# Update linting, formatting

#### Pre-Flight checklist


#### What are the relevant tickets?
#1829 , #1832 

#### What's this PR do?
Currently, if we update Typescript, we will get warnings like
>```
>WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript->eslint/typescript-estree.
>
>You may find that it works just fine, or you may not.
>
>SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.9.0
>
>YOUR TYPESCRIPT VERSION: 4.9.5
>
>Please only submit bug reports when using the officially supported version.
>```

To resolve this warning (and to use new TS features like [`satisfies`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator)) we need to update Prettier from v1 to v2.

Unfortunately... Prettier changed how it handles chained method calls from v1 to v2. See https://github.com/prettier/prettier/issues/7884 for details. _IMO the v1 way looked better for fluent APIs like enzyme has, but...whatever. We should stop using Enzyme anyway._

#### How should this be manually tested?

The only changes here are autofixes by Prettier v2. There should be no behavior change.

So... run Studio and make sure it works. Be sure to either `docker compose up` or manually `docker exec watch yarn install`
